In [1]:
%matplotlib inline
import numpy as np
#import scipy as sp
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split
import pandas as pd
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score

In [2]:
brooklyn=pd.read_csv('brooklyn.csv', low_memory=False)
queens=pd.read_csv('queens.csv', low_memory=False)
brooklyn_homes=pd.read_csv('brooklyn_homes.csv')
queens_homes=pd.read_csv('queens_homes.csv')
queens_low_val=pd.read_csv('queens_low_val.csv')
brooklyn_low_val=pd.read_csv('brooklyn_low_val.csv')
queens=queens.dropna()
brooklyn=brooklyn.dropna()
queens=queens.replace(' -   ','')
queens=queens.replace(' -   ','')

## Logistic Regression

In [3]:
Xqlr, Xqtestlr, yqlr, yqtestlr=train_test_split(queens[['sale_price_clean','year']].values, queens['neighborhood'].values, random_state=5)

In [4]:
clf=LogisticRegression()
clf.fit(Xqlr, yqlr)
print('Accuracy Score For Queens:', accuracy_score(clf.predict(Xqtestlr), yqtestlr))

Accuracy Score For Queens: 0.08261168384879725


This classification model is only 8.26% accurate in predicting the neighborhood in which a sale took place.

In [46]:
Xqhlr, Xqhtestlr, yqhlr, yqhtestlr=train_test_split(queens_homes[['sale_price_clean','year']].values, queens_homes['neighborhood'].values, random_state=5)

In [48]:
clf.fit(Xqhlr, yqhlr)
print('Accuracy Score For Queens Homes:', accuracy_score(clf.predict(Xqhtestlr), yqhtestlr))

Accuracy Score For Queens Homes: 0.11043753294675804


The model is marginally better at 11.043% correct for homes sold in Queens

In [5]:
Xblr, Xbtestlr, yblr, ybtestlr=train_test_split(brooklyn[['sale_price_clean','year']].values, brooklyn['neighborhood'].values, random_state=5)

In [6]:
clf.fit(Xblr, yblr)
print('Accuracy Score for Brooklyn:', accuracy_score(clf.predict(Xbtestlr), ybtestlr))

Accuracy Score for Brooklyn: 0.09866848753840901


The accuracy score for Brooklyn is marginally better at 9.86% but that is still not acceptable.  Perhaps a model to predict the class of building sold could be more accurate

In [49]:
Xbhlr, Xbhtestlr, ybhlr, ybhtestlr=train_test_split(brooklyn_homes[['sale_price_clean','year']].values, brooklyn_homes['neighborhood'].values, random_state=5)

In [50]:
clf.fit(Xbhlr, ybhlr)
print('Accuracy Score For Brooklyn Homes:', accuracy_score(clf.predict(Xbhtestlr), ybhtestlr))

Accuracy Score For Brooklyn Homes: 0.10437956204379562


Again, the model is better for homes than the whole borough, but 10.43% is not acceptable.

In [7]:
Xqlr2, Xqtestlr2, yqlr2, yqtestlr2=train_test_split(queens[['sale_price_clean','year']].values, queens['building_class_category'].values, random_state=5)

In [8]:
clf.fit(Xqlr2, yqlr2)
print('Accuracy Score for Queens Building Category:', accuracy_score(clf.predict(Xqtestlr), yqtestlr2))

Accuracy Score for Queens Building Category: 0.518487972508591


Although the score is much improved from the Neighborhood model (51.84% correct), it is still almost a 50-50 chance of getting the building category correct.

In [9]:
Xblr2, Xbtestlr2, yblr2, ybtestlr2=train_test_split(brooklyn[['sale_price_clean','year']].values, brooklyn['building_class_category'].values, random_state=5)

In [10]:
clf.fit(Xblr2, yblr2)
print('Accuracy Score for Brooklyn Building Category:', accuracy_score(clf.predict(Xbtestlr), ybtestlr2))

Accuracy Score for Brooklyn Building Category: 0.4204506657562308


The accuracy score for Brooklyn shows that the model is not as good at predicting building categories in Brooklyn, as it is in Queens.  42.04% is not a good accuracy score but it is better than 9.86% accuracy score for predicting Neighborhoods in Brooklyn.

In [11]:
Xqlr3, Xqtestlr3, yqlr3, yqtestlr3=train_test_split(queens[['sale_price_clean','gross_square_feet_c']].values, queens['year'].values, random_state=5)

In [12]:
clf.fit(Xqlr3, yqlr3)
print('Accuracy Score for Year of Sale In Queens:', accuracy_score(clf.predict(Xqtestlr3), yqtestlr3))

Accuracy Score for Year of Sale In Queens: 0.7499656357388316


In [13]:
Xblr3, Xbtestlr3, yblr3, ybtestlr3=train_test_split(brooklyn[['sale_price_clean','gross_square_feet_c']].values, brooklyn['year'].values, random_state=5)

In [14]:
clf.fit(Xblr3, yblr3)
print('Accuracy Score for Accuracy Score for Year of Sale In Brooklyn:', accuracy_score(clf.predict(Xbtestlr3), ybtestlr3))

Accuracy Score for Accuracy Score for Year of Sale In Brooklyn: 0.7606691703653123


The model is much better at predicting the year a sale took place in.  The year of a sale was correctly predicted at around 75% for both boroughs.

## Linear Regression

In [15]:
from statsmodels.formula.api import ols

In [16]:
import statsmodels.api as sm

C:\Users\Ryan May\Anaconda3\lib\site-packages\statsmodels\compat\pandas.py:56: FutureWarning: The pandas.core.datetools module is deprecated and will be removed in a future version. Please use the pandas.tseries module instead.
  from pandas.core import datetools


In [17]:
mq=ols('sale_price_clean~gross_square_feet', queens).fit()
print(mq.summary())

                            OLS Regression Results                            
Dep. Variable:       sale_price_clean   R-squared:                       0.044
Model:                            OLS   Adj. R-squared:                  0.044
Method:                 Least Squares   F-statistic:                     1342.
Date:                Thu, 17 May 2018   Prob (F-statistic):          3.35e-287
Time:                        17:14:40   Log-Likelihood:            -4.7328e+05
No. Observations:               29099   AIC:                         9.466e+05
Df Residuals:                   29097   BIC:                         9.466e+05
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
Intercept          5.707e+05   1.65e+0

Above is the linear regression table for the sale price and the gross square feet of sales in Queens.  The coefficent of 17.1496 means that for each additional square foot generates and additional $17.1496 value on the property. 

In [18]:
mb=ols('sale_price_clean~gross_square_feet', brooklyn).fit()
print(mb.summary())

                            OLS Regression Results                            
Dep. Variable:       sale_price_clean   R-squared:                       0.147
Model:                            OLS   Adj. R-squared:                  0.147
Method:                 Least Squares   F-statistic:                     4051.
Date:                Thu, 17 May 2018   Prob (F-statistic):               0.00
Time:                        17:14:40   Log-Likelihood:            -3.8562e+05
No. Observations:               23431   AIC:                         7.712e+05
Df Residuals:                   23429   BIC:                         7.713e+05
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
Intercept          4.582e+05   2.31e+0

The coeffiencient for Brooklyn is quite higher at $99.4183, meaning that an additional square foot in Brooklyn is worth 5.79 times more than an additional square foot in Queens.

In [19]:
from sklearn.linear_model import LinearRegression

In [45]:
queens['n_dummies']=pd.get_dummies(queens['neighborhood'])

ValueError: Wrong number of items passed 59, placement implies 1

In [23]:
pd.get_dummies(queens)

Unnamed: 0  Unnamed: 0.1  block  borough  gross_square_feet  \
1               1             1    949        4               1224   
2               2             2    949        4               1224   
4               4             4    976        4               1224   
6               6             6    976        4               1836   
8               8             8    949        4               2835   
9               9             9    949        4               2635   
20             20            20  15828        4               1200   
21             21            21  15828        4               1200   
22             22            22  15829        4               1650   
23             23            23  15829        4               1650   
24             24            24  15829        4               1650   
25             25            25  15830        4                480   
26             26            26  15830        4                608   
27             27            27  15834        4               1644   
28             28            28  15835        4               2010   
29             29            29  15835        4               2490   
32             32            32  15838        4                926   
35             35            35  15850        4                885   
36             36            36  15850        4                450   
38             38            38  15850        4                450   
39             39            39  15850        4                450   
40             40            40  15890        4               2136   
41             41            41  15890        4               2136   
42             42            42  15907        4               1632   
43             43            43  15908        4               1998   
45             45            45  15916        4               1542   
46             46            46  15952        4               1260   
47             47            47  15952        4                584   
48             48            48  15952        4                584   
49             49            49  15954        4               1408   
...           ...           ...    ...      ...                ...   
66023       66023         66023   1343        4               2850   
66024       66024         66024   2287        4               2920   
66027       66027         66027   2319        4               4149   
66029       66029         66029   2319        4               4872   
66030       66030         66030   2327        4               3705   
66050       66050         66050   1238        4               5208   
66053       66053         66053   1298        4               3364   
66055       66055         66055   1299        4               2480   
66056       66056         66056   1299        4               5156   
66058       66058         66058   1307        4               4000   
66063       66063         66063   1315        4               5850   
66064       66064         66064   1315        4               5616   
66067       66067         66067   1317        4               9800   
66069       66069         66069   1321        4               3680   
66072       66072         66072   2313        4               3600   
66073       66073         66073   2313        4               3600   
66075       66075         66075   1297        4              41558   
66076       66076         66076   1315        4              38813   
66273       66273         66273   1202        4               4196   
66286       66286         66286   1300        4               5220   
66288       66288         66288   1194        4              20000   
66289       66289         66289   1224        4               3750   
66290       66290         66290   1294        4               4900   
66297       66297         66297   1161        4               6500   
66303       66303         66303   1321        4               4000   
66304       66304

In [21]:
lm=LinearRegression()

In [42]:
X=pd.get_dummies(queens.neighborhood)

In [43]:
lm.fit(X,queens.sale_price_clean)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=1, normalize=False)

In [44]:
print('Estimated Intercept Coefficient:' .format(lm.intercept_))

Estimated Intercept Coefficient:
